Feature/hafs docs

mrinalbiswas opened the pull request:
## Description of changes
This PR will update the HAFS documentation on develop.  The "quick start" Users guide covers the basics of checking out the code, compiling, running RTs and some other relevant info a new user will need to start running HAFS on NOAA platforms and MSU's Orion. 

## Issues addressed (optional)
If this PR addresses one or more issues, please provide link(s) to the issue(s) here.
- fixes hafs-community/HAFS/issues/223

## Contributors (optional)
@briannen, @KathrynNewman, @gspetro-NOAA 

## Tests conducted
Regression tests not needed.

BinLiu-NOAA committed:
Add ReadTheDoc related infrastructure, mainly adapted from ufs-weather-model.

briannen committed:
Adding all documentation content from the Google Docs to the Users Guide

briannen committed:
Updating CDEPS workflow images and GitHub branch location from feature to develop

briannen committed:
Updating chapter order

briannen committed:
Fixing spacing between bulleted list items

briannen committed:
Fixing spacing between bulleted list items

briannen committed:
Fixing spacing between bulleted list items

briannen committed:
Fixing spacing between bulleted list items

briannen committed:
Fixing spacing between bulleted list items

briannen committed:
Adding introduction

briannen committed:
Removing broken links

briannen committed:
Updating image sources

briannen committed:
Fixing text on Introduction page

briannen committed:
Removing dates from links

briannen committed:
Fixing broken link from GitVersion

briannen committed:
Adding link to UPP readthedocs

briannen committed:
Adding comments to which links require NOAA email access

mrinalbiswas committed:
Added comments for RT status check

briannen committed:
Enhancing Quick Start guide

KathrynNewman commented:
@gspetro-NOAA (and others), here is the link to the rendered version: https://hafs.readthedocs.io/en/latest/Introduction.html

gspetro-NOAA commented:
@BinLiu-NOAA Sorry I'm just getting to this now! Still looking it over, but one suggestion came to mind right away: Since GitHub doesn't deal well with nonbinary files (they tend to blow up the size of the Git history), the SRW App has instituted a policy where nonbinary files like PNGs are hosted on the repository's wiki (which is its own separate repository). The documentation points to these files when rendering the image, and any changes to these images don't have an effect on the main repository's Git history. Is this a policy HAFS might be interested in implementing? If so, I'm happy to help, since I take care of this on the SRW App side.

gspetro-NOAA reviewed (changes requested):
Based on today's meeting, we need to move the PNGs so that they are hosted on the wiki instead of in the main codebase. I will provide additional info in a separate comment on this PR.

gspetro-NOAA commented:
### Info for Hosting Documentation PNGs on the HAFS Wiki:

The HAFS wiki is its own separate repository. It can be cloned and modified like any repo:
``` 
git clone https://github.com/hafs-community/HAFS.wiki.git
```
The main branch is called `master` by default, and those with permissions to edit the repo wiki should be able to add the PNGs to their local copy of the wiki and commit/push changes directly to the wiki (no PR required). 

You can create an `images` directory in the wiki repo and add the PNGs there; they will be accessible at `https://github.com/hafs-community/HAFS/wiki/images/[NAME_OF_IMAGE_FILE].png`. This URL replaces the file path in the `.. figure::` directive. For example, line #7 of the `Workflow.rst` file would change from `.. figure:: images/hafs_workflow_schematic.png` to: 
``` 
.. figure:: https://github.com/hafs-community/HAFS/wiki/images/hafs_workflow_schematic.png
```

You can see examples of this syntax in the SRW App docs (note that in the SRW App, the PNGs currently sit inside the top-level repo, not in an `images` directory, but the concept is the same, and having an `images` directory is probably better in terms of organizing the wiki contents). For example, the raw code for the SRW App chapter on Limited Area Model grids is [here](https://raw.githubusercontent.com/ufs-community/ufs-srweather-app/develop/docs/UsersGuide/source/CustomizingTheWorkflow/LAMGrids.rst). The rendered docs are [here](https://ufs-srweather-app.readthedocs.io/en/develop/CustomizingTheWorkflow/LAMGrids.html). The `.. figure::` commands in the raw code are responsible for the maps in the documentation. :)

gspetro-NOAA commented on docs/UsersGuide/source/QuickStart.rst:
```diff
@@ -0,0 +1,118 @@
+
+    cd HAFS/parm
+    cp system.conf.<system> system.conf
+    vi system.conf
```
It may make sense just to instruct users to open the file in the command line with their preferred editor and note that `vi system.conf` is one option. That will make things more flexible, since there's vim, MacPorts, and a host of other tools people use.

mrinalbiswas replied:
@gspetro-NOAA Totally agree. Users can use their choice text editors.

gspetro-NOAA commented on docs/UsersGuide/source/QuickStart.rst:
```diff
@@ -0,0 +1,118 @@
+
+* ``cpu_account``
+
+* ``archive=disk`` Archive location (make sure you have write permission)
```
Does this refer to a specific location where the data files are (e.g., path on HPSS or on disk)? Or is it a general label for where to search (e.g., disk, HPSS, nomads, etc.). In `system.conf.jet`, it lists `archive=hpss:/NCEPDEV/{tape_project}/5year/{ENV[USER]}/{SUBEXPT}/{out_prefix}.tar` rather than `archive=disk`.

mrinalbiswas replied:
@gspetro-NOAA Yes, this should refer to a specific location on HPSS.

gspetro-NOAA commented on docs/UsersGuide/source/QuickStart.rst:
```diff
@@ -0,0 +1,118 @@
+
+.. note::
+
+   On Orion, cron is only available on the orion-login-1 node.
```
It might be helpful to tell users where they can find log files, presuming there are some.

mrinalbiswas replied:
@gspetro-NOAA The different task log files (e.g. hafs_atm_lbc1.log   hafs_atm_prep.log  hafs_ocn_post.log etc etc) can be found at CDSCRUB/SUBEXPT/YYYYMMDD/SID. The CDSCRUB is defined in system.conf and SUBEXPT in the cronjob scripts. The users can find other lof files (e.g. hafs_completion_13L_2020082512.log  hafs_scrub_com_13L_2020082512.log	jlogfile
hafs_donefile_13L_2020082512.log    hafs_scrub_work_13L_2020082512.log	rocoto_2020082512.log) under CDSCRUB/SUBEXPT/log.

gspetro-NOAA commented:
I think it could be a good idea to combine the chapters with lists of useful resources. For example, `CodeManagement.rst`, `GuidanceDocs.rst`, and `UsefulResources.rst` could all go in one chapter called "Reference Material", and the section headings would separate content based on topic. I can do that in the PR that I'm preparing (into hafs_docs) if the HAFS team is amenable to that.

gspetro-NOAA commented on docs/UsersGuide/source/CDEPS.rst:
```diff
@@ -0,0 +1,224 @@
+
+Unless you are working on wcoss_cray, the DAPP keyword should already be set to HAFS-ALL.
+ 
+The remainder of the build process is the same as described in the :ref:`HAFS installation guide <hafs_build>`.
```
The `hafs_build` reference is broken. What page/section should it link to?

gspetro-NOAA commented on docs/UsersGuide/source/Prerequisites.rst:
```diff
@@ -0,0 +1,25 @@
+.. _Prerequisites:
+
+**************************************************
+Prerequisites for Working with the HAFS Repository
```
Is HPC-Stack/spack-stack required? For UFS repositories, users generally need more than just Git.

BinLiu-NOAA replied:
I believe so. Basically, HAFS needs majority of the modules/libraries listed in, e.g., https://github.com/hafs-community/HAFS/blob/develop/modulefiles/hafs.orion.lua

gspetro-NOAA commented on docs/UsersGuide/source/CDEPS.rst:
```diff
@@ -0,0 +1,224 @@
+
+The cycle (e.g., 2019082900) and storm (e.g., 00L) can be modified. The final two files in each command configure the CDEPS data models (see :ref:`appendix-a`). It is probably not necessary to change the configuration unless you want to customize the experiment.
+
+Before submitting the cron script, remember to create the **./parm/system.conf** file and to link the fix files using **./sorc/link_fix.sh**. This process is described by the HAFS Developers Guide. (*come back and link to the correct portion of the developers guide*)
```
Do users have to run this script separately? It sounds like `install_hafs.sh` calls the link script when building the app, but it would be good to clarify whether users need to rerun the script when adding new data.

BinLiu-NOAA replied:
@gspetro-NOAA, Typically, ./install_hafs.sh will call and run these steps. However, in situations needed, users can also run these steps manually.

gspetro-NOAA commented on docs/UsersGuide/source/tables/hafs_cdeps_config.csv:
```diff
@@ -0,0 +1,35 @@
+,run_gsi_vr_fgat,Whether to run the GSI-based vortex relocation with FGAT. Must be no if run_datm=yes.
+,run_gsi_vr_ens,Whether to run the GSI-based vortex relocation for each HAFS ensemble member. Must be no if run_datm=yes.
+,run_gsi,Whether to run data assimilation with GSI. Must be no if run_datm=yes.
+,run_fgat,Whether to run data assimilation using FGAT. Must be no if run_datm=yes.
```
What is FGAT?

mrinalbiswas replied:
@gspetro-NOAA FGAT is First-Guess at the Appropriate Time. You can get some idea from [HWRF Users Guide](https://dtcenter.org/sites/default/files/community-code/hwrf/docs/scientific_documents/HWRFv4.0a_ScientificDoc.pdf) (see page 30). There may be some HAFS related documentation which I am unaware.

gspetro-NOAA committed:
Update HAFS documentation (#221)

This PR edits and reorganizes the HAFS documentation. It does not affect HAFS code. 
Summary of edits:
* Removed .pngs files and changed file paths to point to the image locations on the HAFS wiki.
* Created a Technical Overview chapter that combines contents from the former "Prerequisites," "Repository Management," "Submodules," and "Workflow" chapters, which have been removed.
* Created a Reference Materials chapter that combines contents from the former "Guidance Documents," "Code Management," and "Useful Resources" chapters.
* Renamed BasicSteps.rst to GitHubBasicSteps.rst file and edited content.
* Added Glossary chapter (based on Glossary from other UFS apps).
* General edits throughout for clarity, word choice, grammar/punctuation.
* Created empty references.bib file to eliminate warnings (and potentially hold citation info in the future).
* Added Valid Values column to hafs_cdeps_config table.

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/origin/develop' into feature/hafs_docs

BinLiu-NOAA commented:
I have just synced this PR with the latest HAFS develop branch. Please help to go over this PR one more time if you get a chance. After that, this PR is ready to merge (will choose the squash and merge method). Thanks!

gspetro-NOAA commented:
@BinLiu-NOAA @mrinalbiswas It would be nice to have answers to the questions/comments I added to this PR. We can merge without making updates, but as I go through and add to the documentation in the future, this is information that I will need.

BinLiu-NOAA reviewed (approved):
Looks good to me.

BinLiu-NOAA commented:
@gspetro-NOAA, I have added some responses to some of your comments. Please feel free to note there are more questions need to addressed. Meanwhile, I re-requested your review for the latest changes. Thanks!

gspetro-NOAA commented:
@mrinalbiswas @BinLiu-NOAA Thanks for answering my questions! Just approved the PR. I'll add this information to a future PR, so I'm leaving the questions unresolved for now, but feel free to merge. :)
